src/customTypes.d.ts

```diff
@@ -63,3 +63,22 @@ declare module 'mozlog' {
 interface Window {
   gtag: any
 }
+
+declare namespace breach {
```

Comment:
Sample data: 

```
{
  "Name": "000webhost",
  "Title": "000webhost",
  "Domain": "000webhost.com",
  "BreachDate": "2015-03-01",
  "AddedDate": "2015-10-26T23:35:45Z",
  "ModifiedDate": "2017-12-10T21:44:27Z",
  "PwnCount": 14936670,
  "Description": "In approximately March 2015, the free web hosting provider <a href=\"http://www.troyhunt.com/2015/10/breaches-traders-plain-text-passwords.html\" target=\"_blank\" rel=\"noopener\">000webhost suffered a major data breach</a> that exposed almost 15 million customer records. The data was sold and traded before 000webhost was alerted in October. The breach included names, email addresses and plain text passwords.",
  "LogoPath": "https://haveibeenpwned.com/Content/Images/PwnedLogos/000webhost.png",
  "DataClasses": [
    "Email addresses",
    "IP addresses",
    "Names",
    "Passwords"
  ],
  "IsVerified": true,
  "IsFabricated": false,
  "IsSensitive": false,
  "IsRetired": false,
  "IsSpamList": false,
  "IsMalware": false
},
```

Reply:
Verified w/ the above namespace schema and Joi. Doesn't look like any breaches have any more/fewer properties.

```js
import Joi from "joi";

const { array, boolean, number, string } = Joi.types();
const schema = Joi.object({
  Name: string,
  Title: string,
  Domain: string.allow(''),
  BreachDate: string,
  AddedDate: string,
  ModifiedDate: string,
  PwnCount: number,
  Description: string,
  LogoPath: string,
  DataClasses: array.items(string).min(1),
  IsVerified: boolean,
  IsFabricated: boolean,
  IsSensitive: boolean,
  IsRetired: boolean,
  IsSpamList: boolean,
  IsMalware: boolean,
});

try {
  const breaches = await fetchBreaches();
  // delete breaches[5].DataClasses; // Throws an error due to now missing property.
  const opts = { abortEarly: false, presence: 'required' };
  await array.items(schema).validateAsync(breaches, opts);
} catch (err) {
  console.error(err?.details || err);
}

async function fetchBreaches() {
  const res = await fetch("https://haveibeenpwned.com/api/v3/breaches");
  return res.json();
}
```